$NetBSD: patch-aa,v 1.5 2024/12/07 19:42:15 rhialto Exp $

- try to defang the building for 32 bits
- use our $LDFLAGS
- portability fix of `test ==`
- enable version numbered shared library on {Net,Open,Free}BSD

--- configure.orig	2022-08-10 13:33:51.000000000 +0800
+++ configure	2024-12-06 11:33:48.217867627 +0800
@@ -3507,8 +3507,8 @@
 #
 # Now do platform specific tests
 #
-gcc_64bit="-m64"
-gcc_32bit="-m32"
+gcc_64bit=""			# "-m64"
+gcc_32bit="-mfail:nosuchoption"	# "-m32"
 on_osx="no"
 osis64bit=no
 bitflag="32"
@@ -3564,7 +3564,7 @@
          MACH_ARCH="`cat /etc/apk/arch`"
       fi
       ;;
-   *-freebsd* | *-openbsd*)
+   *-freebsd* | *-openbsd* | *-netbsd*)
       mach="`uname -m`"
       if test "$mach" = "amd64"; then
          bitflag="64"
@@ -9040,21 +9040,14 @@
       SHLPRE="lib"
       SHL_LD="ld -assert pure-text -o ${SHLPRE}${SHLFILE}${SHLPST} "'$('SHOFILES')'
       ;;
-   *-freebsd* | *openbsd*)
-      if test "$ac_cv_prog_CC" = "gcc" -o "$ac_cv_prog_CC" = "g++" -o "$ac_cv_prog_CC" = "clang"; then
-         LD_RXLIB_A1="$ac_cv_prog_CC -shared ${LDFLAGS} -o \$(@)"
-         LD_RXLIB_A2="$ac_cv_prog_CC -shared ${LDFLAGS} -o \$(@)"
-         LD_RXLIB_UTILA="$ac_cv_prog_CC -shared ${LDFLAGS} -o \$(@)"
-         SHL_LD="$ac_cv_prog_CC -shared ${LDFLAGS} -o ${SHLPRE}${SHLFILE}${SHLPST} "'$('SHOFILES')'
-      else
-         LD_RXLIB_A1="ld -Bdynamic -Bshareable ${LDFLAGS} -o \$(@)"
-         LD_RXLIB_A2="ld -Bdynamic -Bshareable ${LDFLAGS} -o \$(@)"
-         LD_RXLIB_UTILA="ld -Bdynamic -Bshareable ${LDFLAGS} -o \$(@)"
-         LD_RXLIB_B1="-lc -L. -l${SHLFILE}"
-         LD_RXLIB_B2="-lc -L. -l${SHLFILE}"
-         LD_RXLIB_UTILB="-lc -L. -l${SHLFILE}"
-         SHL_LD="ld -Bdynamic -Bshareable ${LDFLAGS} -o ${SHLPRE}${SHLFILE}${SHLPST} "'$('SHOFILES')'
-      fi
+   *-freebsd* | *-openbsd* | *-netbsd*)
+      LD_RXLIB_A1="${CC} -shared ${LDFLAGS} -o \$(@)"
+      LD_RXLIB_A2="${CC} -shared ${LDFLAGS} -o \$(@)"
+      LD_RXLIB_UTILA="${CC} -shared ${LDFLAGS} -o \$(@)"
+      SHL_LD="${CC} -o ${SHLPRE}${SHLFILE}${SHLPST}.\$(ABI) -shared ${SHL_SCRIPT} -Wl,-soname=${SHLPRE}${SHLFILE}${SHLPST}.\$(ABI_MAJOR) \$(SHOFILES) -lc"
+      SHL_BASE="${SHLPRE}${SHLFILE}${SHLPST}.\$(ABI)"
+      OTHER_INSTALLS="installabilib"
+      USE_ABI="yes"
       STATIC_LDFLAGS="-static"
       SHLPRE="lib"
       ;;
@@ -9252,13 +9245,13 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; } && test -s conftest.o; then
-   mh_dyn_link='ld -shared -o conftest.so.1.0 conftest.o -lc 1>&5'
+   mh_dyn_link='${CC} ${LDFLAGS} -shared -o conftest.so.1.0 conftest.o -lc 1>&5'
    if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$mh_dyn_link\""; } >&5
   (eval $mh_dyn_link) 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; } && test -s conftest.so.1.0; then
-      SHL_LD="ld -shared -o ${SHLPRE}${SHLFILE}${SHLPST} "'$('SHOFILES')'" -lc"
+      SHL_LD="${CC} ${LDFLAGS} -shared -o ${SHLPRE}${SHLFILE}${SHLPST} "'$('SHOFILES')'" -lc"
    else
       mh_dyn_link='ld -G -o conftest.so.1.0 conftest.o 1>&5'
       if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$mh_dyn_link\""; } >&5
@@ -9340,15 +9333,15 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; } && test -s conftest.o; then
-   mh_dyn_link='ld -shared -o conftest.rxlib conftest.o -lc 1>&5'
+   mh_dyn_link='${CC} ${LDFLAGS}  -shared -o conftest.rxlib conftest.o -lc 1>&5'
    if { { eval echo "\"\$as_me\":${as_lineno-$LINENO}: \"$mh_dyn_link\""; } >&5
   (eval $mh_dyn_link) 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; } && test -s conftest.rxlib; then
-      LD_RXLIB_A1="ld -shared -o \$(@)"
-      LD_RXLIB_A2="ld -shared -o \$(@)"
-      LD_RXLIB_UTILA="ld -shared -o \$(@)"
+      LD_RXLIB_A1="${CC} ${LDFLAGS}  -shared -o \$(@)"
+      LD_RXLIB_A2="${CC} ${LDFLAGS}  -shared -o \$(@)"
+      LD_RXLIB_UTILA="${CC} ${LDFLAGS}  -shared -o \$(@)"
       LD_RXLIB_B1="-L. -l${SHLFILE}"
       LD_RXLIB_B2="-L. -l${SHLFILE}"
       LD_RXLIB_UTILB="-L. -l${SHLFILE}"
@@ -9934,7 +9927,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking getting PACKAGE_RELEASE value" >&5
 $as_echo_n "checking getting PACKAGE_RELEASE value... " >&6; }
 PACKAGE_RELEASE=`echo $PACKAGE_RELEASE`
-if test x"$PACKAGE_RELEASE" == "x"; then
+if test x"$PACKAGE_RELEASE" = "x"; then
    PACKAGE_RELEASE=`date +%y%m%d.%H%M`
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PACKAGE_RELEASE" >&5
